lib/rubocop/cop/rails/duplicate_association.rb

```diff
@@ -20,38 +20,91 @@ module Rails
       #   belongs_to :bar
       #   has_one :foo
       #
+      #   # bad
+      #   belongs_to :foo, class_name: 'Foo'
+      #   belongs_to :bar, class_name: 'Foo'
+      #   has_one :baz
```

Comment:
@koic This example (with `belongs_to`) is a false positive. Here's an example that should make this clear:

```
class GraphEdge
  belongs_to :source_node, class_name: "GraphNode"
  belongs_to :target_node, class_name: "GraphNode"
end
```

Contrary to `has_many`, the names of `belongs_to` associations are not arbitrary, as they implicitly reference the foreign key.

Reply:
I noticed the same problem after upgrading to v2.22.0 today.

I think the “identical class_name value” is a valid offense for both `:has_one`, `:has_many` and `:has_and_belongs_to_many` – but not for `:belongs_to`.